task_02/nikolay_shapurov/graph.cpp

```diff
@@ -0,0 +1,137 @@
+#include <array>
+#include <cassert>
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+constexpr int INVALID_ID = -1;
+
+class Graph {
+ private:
+  int num_vertices, num_edges;
+  std::vector<std::vector<int>> data;
+
+ public:
+  Graph(int num_v, int num_e) {
+    assert((num_v >= 0) &&
+           "Invalid number of vertices. Must be positive integer.");
+    assert((num_e >= 0) &&
+           "Invalid number of edges. Must be positive integer.");
+
+    num_vertices = num_v;
+    num_edges = num_e;
+    for (int i = 0; i < num_v; i++) {
+      std::vector<int> temp;
+      data.push_back(temp);
```

Comment:
Для (создания + добавления) объекта в коллекцию существует специальный метод `emplace`, он принимает в себя список аргументов так же как и конструктор:
```suggestion
      data.emplace_back({});
```